File: client/app/src/main/java/ru/spbstu/gyboml/game/GameListener.java
package main.java.ru.spbstu.gyboml.game;

import com.esotericsoftware.kryonet.Connection;
import com.esotericsoftware.kryonet.Listener;

import main.java.ru.spbstu.gyboml.GybomlClient;
import ru.spbstu.gyboml.core.PlayerType;
import ru.spbstu.gyboml.core.net.GameResponses;

public class GameListener extends Listener {

    private Game game;

    GameListener(Game game) { this.game = game; }

    @Override
    public void received(Connection connection, Object object) {
        if (object instanceof GameResponses.Shooted) { shoted(connection, (GameResponses.Shooted)object); }
    }

    private void shoted(Connection connection, GameResponses.Shooted object) {
        PlayerType whoShoted = object.yourShoot ? GybomlClient.getPlayerType() : GybomlClient.getPlayerType().reverted();
        float angle = object.angle;

        synchronized (game) {
            game.physicalScene.generateShot(whoShoted, game.shotType, angle);
            game.soundEffects.shot.play(1.f);
        }
    }
}

File: client/app/src/ru/spbstu/gyboml/game/GameListener.kt
package ru.spbstu.gyboml.game

import com.badlogic.gdx.Gdx
import com.esotericsoftware.kryonet.Connection
import com.esotericsoftware.kryonet.Listener
import ru.spbstu.gyboml.GybomlClient
import ru.spbstu.gyboml.core.net.GameResponses.GameExited
import ru.spbstu.gyboml.core.net.GameResponses.Shooted

class GameListener(private val game: Game) : Listener() {
    override fun received(connection: Connection?, response: Any?) {
        when (response) {
            is Shooted -> shoted(response)
            is GameExited -> exited()
        }
    }

    private fun shoted(response: Shooted) {
        synchronized(game) {
            game.physicalScene.generateShot(game.player.type.reverted(), game.shotType)
            game.physicalScene.lastShot.body.setTransform(response.ballPosition, 0f)
            game.physicalScene.lastShot.velocity = response.ballVelocity
            game.graphicalScene.generateGraphicalShot(game.physicalScene.lastShot)
            game.switchTurn()
        }
    }
    private fun exited() = Gdx.app.exit()
}